packages/dht/src/connection/WebSocket/WebSocketServer.ts

```diff
@@ -22,30 +24,41 @@ declare class NodeJsWsServer extends WsServer { }
 
 export class WebSocketServer extends EventEmitter<ConnectionSourceEvents> {
 
-    private httpServer?: http.Server
+    private httpServer?: HttpServer | HttpsServer
     private wsServer?: WsServer
     private readonly abortController = new AbortController()
-    
-    public async start(portRange: PortRange, host?: string): Promise<number> {
+
+    public async start(portRange: PortRange, host?: string, tlsCertificate?: TlsCertificate): Promise<number> {
         const ports = range(portRange.min, portRange.max + 1)
         for (const port of ports) {
             try {
-                await asAbortable(this.startServer(port, host), this.abortController.signal)
+                await asAbortable(this.startServer(port, host, tlsCertificate), this.abortController.signal)
                 return port
             } catch (err) {
-                logger.debug(`failed to start WebSocket server on port: ${port} reattempting on next port`)
+                if (err.originalError?.code === 'EADDRINUSE') {
+                    logger.debug(`failed to start WebSocket server on port: ${port} reattempting on next port`)
+                } else {
+                    throw new WebSocketServerStartError(err)
+                }
             }
         }
-        throw new StartingWebSocketServerFailed('Failed to start WebSocket server on any port in range')
+        throw new WebSocketServerStartError('Failed to start WebSocket server on any port in range')
     }
 
-    private startServer(port: number, host?: string): Promise<void> {
+    private startServer(port: number, host?: string, tlsCertificate?: TlsCertificate): Promise<void> {
+        const serverCallBack = (request: IncomingMessage, response: ServerResponse<IncomingMessage>) => {
+            logger.trace('Received request for ' + request.url)
+            response.writeHead(404)
+            response.end()
+        }
         return new Promise((resolve, reject) => {
-            this.httpServer = http.createServer((request, response) => {
-                logger.trace('Received request for ' + request.url)
-                response.writeHead(404)
-                response.end()
-            })
+            this.httpServer = tlsCertificate ? 
+                createHttpsServer({
+                    key: fs.readFileSync(tlsCertificate.privateKeyFileName),
+                    cert: fs.readFileSync(tlsCertificate.certFileName)
+                }, serverCallBack)
+                : 
+                createHttpServer(serverCallBack)
```

Comment:
Not related to this PR, but is there a pending feature related to `originIsAllowed`? As it always returns `true`.

Reply:
https://linear.app/streamr/issue/NET-1075/websocket-server-allowed-origins